src/main/java/io/takari/aether/wagon/OkHttpWagon.java

```diff
@@ -332,4 +329,59 @@ public void setHttpHeaders(Properties httpHeaders) {
     this.httpHeaders = Collections.unmodifiableMap(map);
   }
 
+  class InputStreamAdapter extends InputStream {
```

Comment:
This isn't necessary, `response.close()`, `response.body().close()` and `response.body().byteStream().close()` all do the same thing.

If you prefer to keep InputStreamAdapter for readability reasons, please subclass `java.io.FilterInputStream`.